app/assets/javascripts/discourse/app/widgets/header.js

```diff
@@ -315,6 +318,10 @@ export default createWidget("header", {
   tagName: "header.d-header.clearfix",
   buildKey: () => `header`,
 
+  init() {
+    this.appEvents.on("do-not-disturb:changed", this, "scheduleRerender");
```

Comment:
This should be done in the component that attaches the widget, because otherwise the events will not be cleaned up properly.